main.go

```diff
@@ -147,6 +155,92 @@ func doInitrdSetup(ctx *cli.Context) error {
 	return t.InitrdSetup()
 }
 
+func generateKeyPair(newUUID string) error {
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	err = privKey.Validate()
+	if err != nil {
+		return err
+	}
+
+	CN := fmt.Sprintf("manifest PRODUCT:%s", newUUID)
+	template := x509.CertificateRequest {
+		Subject: pkix.Name {
+			CommonName: CN,
+		},
+	}
+
+	// Create a CSR with the new key
+	newCSR, err := x509.CreateCertificateRequest(rand.Reader, &template, privKey)
+	if err != nil {
+		return err
+	}
+
+	keyPEM := pem.EncodeToMemory (
+		&pem.Block {
+			Type: "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(privKey),
+		},
+	)
+
+	csrPEM := pem.EncodeToMemory (
+		&pem.Block {
+			Type: "CERTIFICATE REQUEST",
+			Bytes: newCSR,
+		},
+	)
+
+	//  get trust dir
+	trustDir, err := getTrustPath()
```

Comment:
You probably need to EnsureDir(trustDir) here to make sure the directory exists.  WriteFile() will fail otherwise if you haven't created trustDir elsewhere.

Reply:
ok, thanks!

Reply:
oh wait, getTrustPath() (I added it in utils.go) calls os.MkdirAll(), which should create any missing directories if there are any. Does this seem ok?

Reply:
Yes.